test/validateJoi.js

```diff
@@ -0,0 +1,34 @@
+import chai, { expect } from 'chai';
+import dirtyChai from 'dirty-chai';
+import asPromised from 'chai-as-promised';
+
+import validate from '../src/validateJoi';
+
+chai.use(asPromised);
+chai.use(dirtyChai);
+
+describe('joi validation', () => {
+  context('when the schema is not valid', () => {
+    const schema = {
+      validate: (value, callback) => {
+        const err = new Error();
+        err.name = 'ValidationError';
+
+        return callback(err);
+      },
+    };
+
+    it('rejects', () => expect(validate(schema, null)).to.be.rejected());
+
+    it('sets the error code to 400', () => validate(schema, null)
```

Comment:
It would appear Mr. Henson is right:

> The 422 (Unprocessable Entity) status code means the server understands the content type of the request entity (hence a 415(Unsupported Media Type) status code is inappropriate), and the syntax of the request entity is correct (thus a 400 (Bad Request) status code is inappropriate) but was unable to process the contained instructions.